draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -4621,12 +4621,15 @@ Content-Encoding: gzip
    <x:ref>Range</x:ref> header field in the request (<xref target="field.range"/>).
 </t>
 <t>
-   A client &SHOULD-NOT; generate content in a GET
-   request. Content received in a GET request has no defined semantics,
+   Content received in a GET request has no generally defined semantics,
    cannot alter the meaning or target of the request, and might lead some
    implementations to reject the request and close the connection because of
    its potential as a request smuggling attack
    (<xref target="request.smuggling"/>).
+   A client &SHOULD-NOT; generate content in a GET request unless it is
+   made directly to an origin server that has previously indicated,
+   in or out of band, that such a request has a purpose and will be adequately
+   supported.
```

Comment:
This places a requirement on clients and what they generate, but not servers (and especially resources) regarding what they require. As discussed, the latter are often the real problem. So, suggest:

```suggestion
   A client &SHOULD-NOT; generate content in a GET request unless it is
   made directly to an origin server that has previously indicated,
   in or out of band, that such a request has a purpose and will be adequately
   supported. Because some clients might use proxies or other infrastructure
   outside the control of the server, it is NOT RECOMMENDED that resources
   require the use of content on a GET request to interoperate.
```

... with corresponding edits for `HEAD` and `DELETE`.

Reply:
A double-SHOULD NOT is an improvement.

"resources &SHOULD-NOT; require the use of content" perhaps, just to keep this parallel.

Reply:
By all means, do not include DELETE into this issue right now.

Reply:
@reschke if we want to finish this, it'd really help to know _why_ you object.

Reply:
I'm ok with this, although I don't think it's needed. If we do this, I would prefer "SHOULD NOT", as this would be the first occurrence of "NOT RECOMMENDED" in the spec.

Reply:
We don't make requirements on resources. It would make more sense to say that an origin server cannot rely on private agreements unless they control the entire request/response chain.

Reply:
https://github.com/httpwg/http-core/pull/906/commits/fbdad50aa9bb577b13518a610c12a2fa36d4f0f1